tests/validate-urls.rb

```diff
@@ -15,19 +15,22 @@
 }
 
 # Check if the supplied URL works
+# rubocop:disable Metrics/AbcSize
 def check_url(path, url, res = nil)
   loop do
-    res = Net::HTTP.get_response(URI.parse(url))
+    url = URI.parse(url)
+    res = Net::HTTP.get_response(url)
     break unless res.is_a? Net::HTTPRedirection
 
-    url = res['location']
+    url = URI(res['location']).is_a?(URI::HTTP) ? res['location'] : "#{url.scheme}://#{url.host}#{res['location']}"
   end
 
   puts "::warning file=#{path}:: Unexpected response from #{url} (#{res.code})" unless res.is_a? Net::HTTPSuccess
 rescue StandardError => e
-  puts "::warning file=#{path}:: Unable to reach #{url}"
+  puts "::error file=#{path}:: Unable to reach #{url}"
```

Comment:
I also switched unreachable URLs from outputting warnings to errors.

Reply:
Some sites are only accessible from a certain region (or with a VPN), so a hard error might be a bit much here.

Reply:
Good point :thinking:
I'll revert this line then